VideoLocker/src/main/java/org/edx/mobile/util/FileUtil.java

```diff
@@ -0,0 +1,39 @@
+package org.edx.mobile.util;
+
+import android.content.Context;
+
+import java.io.BufferedInputStream;
+import java.io.ByteArrayOutputStream;
+import java.io.IOException;
+import java.io.InputStream;
+
+/**
+ * Created by miankhalid on 8/3/15.
+ */
+public class FileUtil {
+    /**
+     * Returns the text of a file as a String object
+     * @param context   current context
+     * @param fileName  name of the file to load from assets folder
+     * @return  text content of the file
+     */
+    public static String loadTextFileFromAssets(Context context, String fileName) {
+        String fileContent;
+        try {
+            InputStream iStream = context.getAssets().open(fileName);
+            BufferedInputStream bInputStream = new BufferedInputStream(iStream);
+            ByteArrayOutputStream bOutputStream = new ByteArrayOutputStream();
+            int result = bInputStream.read();
```

Comment:
Based on the `Files.readAllBytes()` method implementation in Java 7, which depends on the file size being known:

    public static String loadTextFileFromAssets(Context context, String fileName) throws IOException {
        AssetManager assets = context.getAssets();
        AssetFileDescriptor fd = assets.openFd(fileName);
        try {
            InputStream in = fd.createInputStream();
            try {
                in = new BufferedInputStream(in);
                // read to EOF which may read more or less than initialSize (eg: file
                // is truncated while we are reading)
                int capacity = (int) fd.getLength();
                byte[] buf = new byte[capacity];
                int nread = 0;
                int rem = buf.length;
                int n;
                while ((n = in.read(buf, nread, rem)) > 0) {
                    nread += n;
                    rem -= n;
                    assert rem >= 0;
                    if (rem == 0) {
                        // need larger buffer
                        int newCapacity = capacity << 1;
                        if (newCapacity < 0) {
                            if (capacity == Integer.MAX_VALUE)
                                throw new OutOfMemoryError("Required array size too large");
                            newCapacity = Integer.MAX_VALUE;
                        }
                        rem = newCapacity - capacity;
                        buf = Arrays.copyOf(buf, newCapacity);
                        capacity = newCapacity;
                    }
                }
                return new String((capacity == nread) ? buf : Arrays.copyOf(buf, nread));
            } finally {
                in.close();
            }
        } finally {
            fd.close();
        }
    }

If we want to handle the case where `getLength()` returns `UNKNOWN_LENGTH`, then it's probably better to use a custom buffer of a reasonable size, such as done in `IOUtils` from Apache Commons:

    byte[] buffer = new byte[1024 * 4];
    int n = 0;
    while (-1 != (n = input.read(buffer))) {
        output.write(buffer, 0, n);
    }

This avoids using the `available()` method, which is quite conservative and offers very weak guarantees.

[AssetFileDescriptor#getLength() documentation]: https://developer.android.com/reference/android/content/res/AssetFileDescriptor.html#getLength()